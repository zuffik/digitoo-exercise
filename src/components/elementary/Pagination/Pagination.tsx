import * as React from 'react';
import ReactPaginate from "react-paginate";
import {Pagination as PaginationDTO} from "../../../services/types/entity/Pagination";

interface Props {
    onOffsetChange: (offset: number, limit: number) => void;
    limit?: number;
    pagination: PaginationDTO;
}

export const Pagination: React.FC<Props> = (props: Props) => {
    const limit = props.limit || 10;
    return (
        <>
            <ReactPaginate previousClassName="page-item"
                           previousLinkClassName="page-link"
                           nextClassName="page-item"
                           nextLinkClassName="page-link"
                           containerClassName="pagination justify-content-center"
                           pageClassName="page-item"
                           pageLinkClassName="page-link"
                           activeClassName="page-item active"
                           activeLinkClassName="page-link"
                           onPageChange={item => props.onOffsetChange(item.selected * limit, limit)}
                           disableInitialCallback
                           pageCount={Math.ceil(props.pagination.total / limit)}
                           marginPagesDisplayed={3}
                           pageRangeDisplayed={3}
                           initialPage={props.pagination.offset / limit}/>
        </>
    );
};
