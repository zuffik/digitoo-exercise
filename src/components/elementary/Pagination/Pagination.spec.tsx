import * as React from 'react';
import {render} from '@testing-library/react'
import {Pagination} from "./Pagination";

describe('<Pagination/>', () => {
    it('should render', () => {
        const {asFragment} = render(<Pagination onOffsetChange={jest.fn()} pagination={{limit:10, offset: 0, total: 100}}/>);
        expect(asFragment()).toMatchSnapshot();
    });
});
