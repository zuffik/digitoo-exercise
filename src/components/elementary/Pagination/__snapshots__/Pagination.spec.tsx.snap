// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination/> should render 1`] = `
<DocumentFragment>
  <ul
    class="pagination justify-content-center"
  >
    <li
      class="page-item disabled"
    >
      <a
        aria-disabled="true"
        aria-label="Previous page"
        class="page-link"
        role="button"
        tabindex="0"
      >
        Previous
      </a>
    </li>
    <li
      class="page-item page-item active"
    >
      <a
        aria-current="page"
        aria-label="Page 1 is your current page"
        class="page-link page-link"
        role="button"
        tabindex="0"
      >
        1
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        aria-label="Page 2"
        class="page-link"
        role="button"
        tabindex="0"
      >
        2
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        aria-label="Page 3"
        class="page-link"
        role="button"
        tabindex="0"
      >
        3
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        aria-label="Page 4"
        class="page-link"
        role="button"
        tabindex="0"
      >
        4
      </a>
    </li>
    <li
      class="break"
    >
      <a
        role="button"
        tabindex="0"
      >
        ...
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        aria-label="Page 8"
        class="page-link"
        role="button"
        tabindex="0"
      >
        8
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        aria-label="Page 9"
        class="page-link"
        role="button"
        tabindex="0"
      >
        9
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        aria-label="Page 10"
        class="page-link"
        role="button"
        tabindex="0"
      >
        10
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        aria-disabled="false"
        aria-label="Next page"
        class="page-link"
        role="button"
        tabindex="0"
      >
        Next
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
